unit-tests/heitech.zer0mqXt.core.tests/RequestReplyTests.cs

```diff
@@ -138,6 +138,47 @@ public async Task AsyncRqRep()
             Assert.True(result.IsSuccess);
         }
 
+        [Fact]
+        public async Task Inheritance_for_req_rep_tested()
+        {
+            // Arrange
+            var configuration = new ConfigurationTestData().GetSocketConfigInProc;
+            configuration.Logger.SetSilent();
+            var sut = new Socket(configuration);
+            sut.Respond<InheritedRequest, InheritedResponse>(rq =>
+            {
+                return new InheritedResponse();
+            });
+
+            // Act
+            var xtResult = await sut.RequestAsync<Request, Response>(new Request());
+
+            // Assert
+            Assert.False(xtResult.IsSuccess);
+            sut.Dispose();
+        }
+
+
+        [Fact]
+        public async Task Inheritance_for_req_rep_tested_on_request_side()
+        {
+            // Arrange
+            var configuration = new ConfigurationTestData().GetSocketConfigInProc;
+            var sut = new Socket(configuration);
+            sut.Respond<Request, Response>(rq =>
+            {
+                return new Response();
+            });
+
+            // Act
+            var xtResult = await sut.RequestAsync<InheritedRequest, InheritedResponse>(new InheritedRequest());
+
+            // Assert
+            Assert.False(xtResult.IsSuccess);
```

Comment:
Is this test expected to fail because the request and response parts don't use the same type?

And shouldn't we also have a positive test, which expects the derived types `ResponseNumber`?